### DISCUSSION TITLE
Error generating bootstrap repo for: SLE-15-SP5-x86_64 : ERROR: package 'venv-salt-minion' not found

### DISCUSSION THREAD

**Discussion by ipanmeri** [uyuni-project/uyuni#8623] [Q&A] [answered]
Hi,
I try to get the bootstrap together, but always have this error:
`Error generating bootstrap repo for: SLE-15-SP5-x86_64 : ERROR: package 'venv-salt-minion' not found`

Not seeing the Uyuni Client Tools listed in the Web UI I tried, following [this docu](https://www.uyuni-project.org/uyuni-docs/en/uyuni/client-configuration/clients-sle.html), to add the repo:
 
uyunisrv:~ # spacewalk-common-channels sles12-sp5-pool-x86_64 sles12-sp5-uyuni-client
SUSE Manager username: xxxxx
SUSE Manager password: xxxxx
INFO: sles12-sp5-pool-x86_64 exists
WARNING: sles12-sp5-uyuni-client-x86_64: redstone.xmlrpc.XmlRpcFault: The channel name 'Uyuni Client Tools for SLES12 SP5 x86_64' is already in use, please enter a different name

So I guess, the repo for the Uyuni Client Tools, that should deliver venv-salt-minion, is in place.
What am I missing?
Thanks a lot

![uu_sles12sp5_venv](https://github.com/uyuni-project/uyuni/assets/161012190/6b27623a-5bf7-4814-9b2a-c3ce441a600e)

**Comment by ipanmeri** [comment]
I just found this log file : sles12-sp5-uyuni-client-x86_64.log
I find lots of download errors like

`2024/04/19 12:58:39 +02:00 ERROR: Download failed: https://download.opensuse.org/repositories/systemsmanagement:/Uyuni:/Stable:/SLE12-Uyuni-Client-Tools/SLE_12/x86_64/wire-debuginfo-0.5.0-9.13.uyuni.x86_64.rpm - [Errno 14] HTTPS Error 302 - Found.
2024/04/19 12:58:39 +02:00     143/150 : wire-debuginfo-0.5.0-9.13.uyuni.x86_64.rpm (failed)
2024/04/19 12:58:39 +02:00 WARNING: Download failed: https://download.opensuse.org/repositories/systemsmanagement:/Uyuni:/Stable:/SLE12-Uyuni-Client-Tools/SLE_12/x86_64/zeromq-debugsource-4.0.4-15.8.13.uyuni.x86_64.rpm - [Errno 14] HTTPS Error 302 - Found. Retrying...
2024/04/19 12:58:39 +02:00 WARNING: Download failed: https://download.opensuse.org/repositories/systemsmanagement:/Uyuni:/Stable:/SLE12-Uyuni-Client-Tools/SLE_12/x86_64/zeromq-debugsource-4.0.4-15.8.13.uyuni.x86_64.rpm - [Errno 14] HTTPS Error 302 - Found. Retrying...
`

I then tried to download one of the packages on a windows chrome and got the pkg.
Then I tried to download the pkg via wget onto the uyuni server and got:

`uyunisrv:/tmp # wget https://download.opensuse.org/repositories/systemsmanagement:/Uyuni:/Stable:/SLE12-Uyuni-Client-Tools/SLE_12/x86_64/zeromq-debugsource-4.0.4-15.8.13.uyuni.x86_64.rpm
--2024-04-19 13:21:45--  https://download.opensuse.org/repositories/systemsmanagement:/Uyuni:/Stable:/SLE12-Uyuni-Client-Tools/SLE_12/x86_64/zeromq-debugsource-4.0.4-15.8.13.uyuni.x86_64.rpm
Resolving download.opensuse.org (download.opensuse.org)... 195.135.223.226, 2a07:de40:b250:131:10:151:131:30
Connecting to download.opensuse.org (download.opensuse.org)|195.135.223.226|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://opensuse.mirror.liquidtelecom.com/repositories/systemsmanagement%3A/Uyuni%3A/Stable%3A/SLE12-Uyuni-Client-Tools/SLE_12/x86_64/zeromq-debugsource-4.0.4-15.8.13.uyuni.x86_64.rpm [following]
--2024-04-19 13:21:46--  https://opensuse.mirror.liquidtelecom.com/repositories/systemsmanagement%3A/Uyuni%3A/Stable%3A/SLE12-Uyuni-Client-Tools/SLE_12/x86_64/zeromq-debugsource-4.0.4-15.8.13.uyuni.x86_64.rpm
Resolving opensuse.mirror.liquidtelecom.com (opensuse.mirror.liquidtelecom.com)... 197.155.77.1, 2c0f:fe40:8001:10::1
Connecting to opensuse.mirror.liquidtelecom.com (opensuse.mirror.liquidtelecom.com)|197.155.77.1|:443... connected.
ERROR: cannot verify opensuse.mirror.liquidtelecom.com's certificate, issued by ‘emailAddress=support@sophos.com,CN=Sophos SSL CA_hkfCOnyRr0uIznc,OU=NSG,O=Sophos,ST=Oxfordshire,C=GB’:
  Self-signed certificate encountered.
To connect to opensuse.mirror.liquidtelecom.com insecurely, use `--no-check-certificate'.
uyunisrv:/tmp #
`
Is this ok, I mean, are pkgs hosted on other platforms/site also?
Thanks

**Reply by ipanmeri** [reply]
using the --no-check-certificate with wget I can download the pkg.

**Reply by agraul** [reply, marked as answer]
>  Is this ok, I mean, are pkgs hosted on other platforms/site also?

download.opensuse.org is a mirror-redirector. Sadly it sometimes redirects to broken mirrors and you get error like this. `--no-check-certificate` should not be needed. The best you can do is to sync the channel again.

**Comment by ipanmeri** [comment]
I suppose this is related: what does it mean or better, how do I fix
"no access to parent channel"
for the Uyuni Client Tools for SLES12 SP5 x86_64 ?

![uu_noaccess](https://github.com/uyuni-project/uyuni/assets/161012190/0437af4e-8a77-421c-ac02-2e6345058674)

Thanks
